src/utils/animations.ts

```diff
@@ -1,26 +1,28 @@
-import { RefObject } from 'react'
+import { RefObject, useEffect } from 'react'
 
 // Some tests will pass undefined for the Document.
 export function isAnimating(node?: Animatable | Document) {
   return (node?.getAnimations?.().length ?? 0) > 0
 }
 
-export const UNMOUNTING = 'unmounting'
-
 /**
  * Delays a node's unmounting until any animations on that node are finished, so that an unmounting
  * animation may be applied. If there is no animation, this is a no-op.
  *
- * CSS should target the UNMOUNTING class to determine when to apply an unmounting animation.
+ * CSS should target the animationClass to determine when to apply the animation. If animationClass
+ * is a function, it will be invoking when the node would begin unmounting.
  */
-export function delayUnmountForAnimation(node: RefObject<HTMLElement>) {
-  const current = node.current
-  const parent = current?.parentElement
-  const removeChild = parent?.removeChild
-  if (parent && removeChild) {
+export function useUnmountingAnimation(node: RefObject<HTMLElement>, animatingClass: (() => string) | string) {
+  useEffect(() => {
+    const current = node.current
+    const parent = current?.parentElement
+    const removeChild = parent?.removeChild
+    if (!(parent && removeChild)) return
+
     parent.removeChild = function <T extends Node>(child: T) {
       if ((child as Node) === current) {
-        current.classList.add(UNMOUNTING)
+        const klass = typeof animatingClass === 'string' ? animatingClass : animatingClass()
```

Comment:
not sure i'm following here. why could animating class be a class or string?

Reply:
I've narrowed the type (to a callback) and tried to explain why this is a callback (and not a string) through comments.